libs/arcade-tdk/arcade_tdk/providers/http/error_adapter.py

```diff
@@ -109,8 +110,10 @@ def _is_rate_limit_403(self, headers: dict[str, str], msg: str) -> bool:
         Returns:
             True if this 403 should be treated as rate limiting
         """
-        # Check if any rate limiting headers are present
-        return any(header.lower() in [h.lower() for h in headers] for header in RATE_HEADERS)
+        # Check if rate limit is actually exhausted (remaining requests is 0)
+        headers_lower = {k.lower(): v for k, v in headers.items()}
+        remaining = headers_lower.get("x-ratelimit-remaining")
```

Comment:
`x-ratelimit-remaining` is probably not universal (?). Quick search for equivalent headers could help us make this more robust `ratelimit-remaining`, `x-rate-limit-remaining`, `x-app-rate-limit-remaining`, etc.